File: download/src/main/java/com/example/download/db/ThreadDAO.java
package com.example.download.db;

import com.example.download.bean.ThreadInfo;

import java.util.List;

/**
 * @author ShenBF
 * @描述: 数据访问接口
 * @date 2018/4/21
 */
public interface ThreadDAO {

    /**
     * 插入线程信息
     */
    public void insertThread(ThreadInfo threadInfo);

    /**
     * 删除线程信息
     */
    public void deleteThread(String url, int thread_id);

    /**
     * 更新线程下载进度
     */
    public void updateThread(String url, int thread_id, int finished);

    /**
     * 查询文件的线程信息
     */
    public List<ThreadInfo> getThreads(String url);

    /**
     * 线程信息是否存在
     */
    public boolean isExists(String url, int thread_id);

}

File: library/download/src/main/java/com/example/download/db/ThreadDAO.kt
package com.example.download.db

import com.example.download.bean.ThreadInfo

/**
 * @author fqxyi
 * @描述: 数据访问接口
 * @date 2018/4/21
 */
interface ThreadDAO {
    /**
     * 插入线程信息
     */
    fun insertThread(threadInfo: ThreadInfo)

    /**
     * 删除线程信息
     */
    fun deleteThread(url: String?, thread_id: Int)

    /**
     * 更新线程下载进度
     */
    fun updateThread(url: String?, thread_id: Int, finished: Int)

    /**
     * 查询文件的线程信息
     */
    fun getThreads(url: String?): List<ThreadInfo?>?

    /**
     * 线程信息是否存在
     */
    fun isExists(url: String?, thread_id: Int): Boolean
}